pkg/cli/config.go

```diff
@@ -6,51 +6,35 @@
 package cli
 
 import (
-	"context"
 	"fmt"
+	"os"
+	"strconv"
 
 	"github.com/getoutreach/gobox/pkg/cfg"
-	"github.com/getoutreach/gobox/pkg/secrets"
+	"github.com/getoutreach/gobox/pkg/cli/logfile"
 	"github.com/getoutreach/gobox/pkg/trace"
 	"gopkg.in/yaml.v3"
 )
 
 // overrideConfigLoaders fakes certain parts of the config that usually get pulled
 // in via mechanisms that don't make sense to use in CLIs.
-func overrideConfigLoaders(honeycombAPIKey, dataset string, tracingDebug bool) {
-	// override the secret loader so that we can read specific keys from variables
-	// otherwise fallback to the original secret loader, if it was set.
-	var fallbackSecretLookup func(context.Context, string) ([]byte, error)
-	fallbackSecretLookup = secrets.SetDevLookup(func(ctx context.Context, path string) ([]byte, error) {
-		// use the embedded in value
-		if path == "APIKey" {
-			return []byte(honeycombAPIKey), nil
-		}
-
-		// if no fallback, return an error, failed to find :(
-		// note: as of this time the secrets logic looks for
-		// the path before falling back to the devlookup so this
-		// is safe to assume all attempts have failed
-		if fallbackSecretLookup == nil {
-			return nil, fmt.Errorf("failed to find secret at path '%s', or compiled into binary", path)
-		}
+func overrideConfigLoaders() {
+	fmt.Println("in override config loaders")
 
-		return fallbackSecretLookup(ctx, path)
-	})
+	portStr, _ := os.LookupEnv(logfile.TracePortEnvironmentVariable)
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		panic("invalid tracing port provided")
+	}
 
 	fallbackConfigReader := cfg.DefaultReader()
+	fmt.Println("created fallbackConfigReader")
 	cfg.SetDefaultReader(func(fileName string) ([]byte, error) {
+		fmt.Printf("overriding config: %s/n", fileName)
 		if fileName == "trace.yaml" {
 			traceConfig := &trace.Config{
-				Otel: trace.Otel{
-					Enabled:  true,
-					Endpoint: "api.honeycomb.io",
-					APIKey: cfg.Secret{
-						Path: "APIKey",
-					},
-					Debug:         tracingDebug,
-					Dataset:       dataset,
-					SamplePercent: 100,
+				LogFile: trace.LogFile{
```

Comment:
Maybe this should still be under `otel` since we're technically changing an exporter? Something like that? Not super attached

Reply:
I wanted to support this as a separate config since it's fairly different than the standard otel config. We use our own exporter, skip sampling, and generally don't allow as many configs.